packages/playwright-core/src/server/chromium/crNetworkManager.ts

```diff
@@ -311,7 +311,11 @@ export class CRNetworkManager {
       // right away, so that client can call it from the route handler.
       request.request.setRawRequestHeaders(headersObjectToArray(requestPausedEvent.request.headers, '\n'));
     }
-    (this._page?._frameManager || this._serviceWorker)!.requestStarted(request.request, route || undefined);
+    const dispatchRequestStarted = () => (this._page?._frameManager || this._serviceWorker)!.requestStarted(request.request, route || undefined);
+    if (redirectedFrom)
+      this._responseExtraInfoTracker.dispatchWhenResponseReady(redirectedFrom._requestId, redirectedFrom.request._existingResponse(), dispatchRequestStarted);
```

Comment:
Instead of the callback indirection that elevates the complexity, I would prefer an explicit approach where you pass the event target explicitly in the tracker constructor and let it dispatch these events from its own code. Would be easier to follow.